### DISCUSSION TITLE
LDAP authentication

### DISCUSSION THREAD

**Discussion by Yorny2101** [Mailu/Mailu#3326] [Feature requests / Ideas]
Hi everyone,

I noticed that about six years ago, a solution for integrating LDAP authentication was proposed in PR #715. @muhlemmer pointed out that it might conflict with PR #670, which aimed to change the configuration management. He suggested two options: either base the modifications on PR #670 or review and merge PR #715 as-is and then modify PR #670 accordingly.

Given the current state of Mailu, I'm wondering how I can adapt the original LDAP integration solution from PR #715 to the current codebase. Any guidance or advice on how to proceed would be greatly appreciated.

Thanks in advance for your help!

**Comment by nextgens** [comment]
The author of the PR wrote: "I didn't recognize the complexity of <> this feature." and then subsequently gave up on implementing it.

I am not aware of anyone expressing interest in it since. If what you are after is just authentication, look into [this feature](https://mailu.io/2024.06/configuration.html#header-authentication-using-an-external-proxy) and see if you can get a 3rd party SSO solution to do the interfacing for you without code changes.

If you are serious about trying to implement it in Mailu, start by speccing out what it is that you want (and that wouldn't be provided by what I have hinted at above) and see if there are other people who are interested.

**Comment by Yorny2101** [comment]
Thank you for the response!

Is there any existing example configurations on how to integrate a third-party SSO solution? I am going to proceed using Authentik with an OpenID provider for authentication. Any additional guidance would be greatly appreciated.

**Comment by nobody8v** [comment]
Hey there!

I'm also working on a similar task and would like to learn more about this approach. Is there more information about the setup of the proxy in mailu to get auth in Authentik?
I'm testing everything in a local deployment with Docker, and also would like to know if any component of Mailu should be in the same network of Authentik server?

Thanks in advance!!

**Reply by dasbenjo** [reply]
hey, maybe it's useful, here is my config to use authentik as an identity provider for mailu (make sure to replace the brackets). Just add these variables to mailu.env:

#Proxy (this is authentik)
PROXY_AUTH_WHITELIST=<local ip of authentik server/vm>/32

#header with mail address in authentik
PROXY_AUTH_HEADER=X-authentik-email

#most likely you want this set to true, you can control the access in authentik with groups
PROXY_AUTH_CREATE=true

#Logout URL for authentik proxy
PROXY_AUTH_LOGOUT_URL=https://<mail.domain.mailu>/outpost.goauthentik.io/sign_out

in Authentik, you need to add a proxy (i use nginx with forward auth). The external host is <mail.domain.mailu>

I have seperated docker compose files for both services, because they are on different VMs.

**Reply by nobody8v** [reply]
thanks for your answer!! it's very useful and I think it will help to clarify the settings.
